### ISSUE TITLE
Can't sort Templates or ISOs.

### ISSUE THREAD

**Issue by edikevich** [apache/cloudstack#5731]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
UI
~~~

##### CLOUDSTACK VERSION
~~~
4.16.0
~~~

##### CONFIGURATION
XCP-NG 8.2
Advanced Zone

##### OS / ENVIRONMENT
Oracle Linux 8.4 - Management server, MYSQL 8.0.26
XCP-NG 8.2 - hypervisor

##### SUMMARY
Can't sort Templates or ISOs.
In Templates section buttons are presented but when you try to click them nothing happends.

In ISOs section buttons are missed (were in old UI)

##### STEPS TO REPRODUCE
~~~
Try to sort templates.
~~~
##### EXPECTED RESULTS
~~~
Can place templates in any order
~~~

##### ACTUAL RESULTS
~~~
Can't place templates in any order. When you try to click sort buttons nothing happends. In ISOs section buttons are missed.
~~~

**Comment by sureshanaparti** [comment]
@utchoang can you check this please. thanks.

**Comment by utchoang** [comment]
Through testing, I found the problem is that when getting listTemplates with `showunique: true` it sorts wrong. But when I remove this parameter, the returned data is sorted correctly, but the result is a sequence of duplicate samples. As far as I can see is it coming from the API?
- With `showunique: true`
![image](https://user-images.githubusercontent.com/13766648/146481630-ec075c87-7e46-49d3-8e6a-6aa525e9f175.png)
- Without `showunique: true`
![image](https://user-images.githubusercontent.com/13766648/146481763-8e7c53a9-5cd1-4202-aed3-62d2d6f1ae13.png)

**Comment by utchoang** [comment]
I think with API `updateTemplate /updateIos` to update sortKey need one more parameter to be able to update for duplicate templates/isos?

**Comment by sureshanaparti** [comment]
> Through testing, I found the problem is that when getting listTemplates with `showunique: true` it sorts wrong. But when I remove this parameter, the returned data is sorted correctly, but the result is a sequence of duplicate samples. As far as I can see is it coming from the API?
> 
> * With `showunique: true`
>   ![image](https://user-images.githubusercontent.com/13766648/146481630-ec075c87-7e46-49d3-8e6a-6aa525e9f175.png)
> * Without `showunique: true`
>   ![image](https://user-images.githubusercontent.com/13766648/146481763-8e7c53a9-5cd1-4202-aed3-62d2d6f1ae13.png)

@utchoang With 'showunique: true', API returns unique templates irrespective of zones, and without 'showunique: true', all the templates in the zones are returned. So, if a template is registered for mutiple/all zones, it is present multiple times.
From UI, the templates are always listed with 'showunique: true'.

**Comment by sureshanaparti** [comment]
@utchoang Fixed the API response in the PR #5800.

**Comment by sureshanaparti** [comment]
Fixed with #5800. Closing the issue.
